### PR TITLE
Vi resolution copy

### PR DESCRIPTION
**Pull request by Vilax**

### CODE REVIEW

File: xmipp3/protocols/__init__.py [hunk 1]
```diff
@@ -85,6 +85,7 @@
 from .protocol_reconstruct_swarm import XmippProtReconstructSwarm
 from .protocol_resolution3d import XmippProtResolution3D
 from .protocol_resolution_directional import XmippProtMonoDir
+#from .protocol_resolution_localfilter import XmippProtResLocalFilter
```
Comment (dmaluenda) [inline xmipp3/protocols/__init__.py]: It shouldn't be commented.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        self._insertFunctionStep('localFilterStep')
+        self._insertFunctionStep('createOutputStep')
+
+
+    def convertInputStep(self):
+        """ Read the input volume.
+        """
+
+        self.volFn = self.inputVolume.get().getFileName()
+        self.resFn = self.inputVolume2.get().getFileName()
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: `inputVolume2` doesn't exist. It is `resVol`.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        
+        extVol1 = getExt(self.vol1Fn)
+        extVol2 = getExt(self.vol2Fn)
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: `vol2Fn` doesn't exist, it should be `resFn`.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+    # --------------------------- INSERT steps functions --------------------------------------------
+
+    def _createFilenameTemplates(self):
+        """ Centralize how files are called """
+        myDict = {
+                 FN_MEAN_VOL: self._getExtraPath('mean_volume.vol'),
+                 OUTPUT_MASK_FILE: self._getExtraPath("output_Mask.vol"),
+                 OUTPUT_RESOLUTION_FILE_CHIMERA: self._getExtraPath(CHIMERA_RESOLUTION_VOL),
+                 FN_FILTERED_MAP: self._getExtraPath('filteredMap.vol'),
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: The keys of this dict don't exist.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    def localFilterStep(self):
+        # Number of frequencies
+        max_ = self.maxRes.get()
+
+        if self.stepSize.hasValue():
+            freq_step = self.stepSize.get()
+        else:
+            freq_step = 0.25
+  
+        xdim = self.maskRadius()
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: Code from line 117 to line 124 is not used.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+        params = ' --vol %s' % self.volFn
+        params += ' --resVol %s' % self.resFn
+        params += ' --sampling_rate %f' % self.inputVolume.get().getSamplingRate()
+        params += ' -o %s' % self._getFileName(OUTPUT_RESOLUTION_FILE)
+        params += ' --filteredMap %s' % self._getFileName(FN_FILTERED_MAP)
+        params += ' --threads %f' %
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: Incomplete line. It should be ...`% self.numberOfThreads.get()`

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+       
+        
+            addParamsLine("  --vol <vol_file=\"\">               : Volume");
+    addParamsLine("  --resvol <vol_file=\"\">                : Resolution map");
+    addParamsLine("  -o <output=\"MGresolution.vol\">    : Local resolution volume (in Angstroms)");
+    addParamsLine("  --filteredMap <output=\"filteredMap.vol\">    : Local resolution volume filtered (in Angstroms)");
+    addParamsLine("  [--sampling_rate <s=1>]               : Sampling rate (A/px)");
+    addParamsLine("  [--step <s=0.25>]                   : The resolution is computed at a number of frequencies between minimum and");
+    addParamsLine("  [--significance <s=0.95>]           : The level of confidence for the hypothesis test.");
+    addParamsLine("  [--threads <s=4>]                   : Number of threads");
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: All this is from the C++ code. Even thought, it is very useful to have it here in order to know which parameters are able, it should be in a comment. Keep the indention!

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+        self.runJob('xmipp_resolution_localfilter', params)
+
+
+    def createOutputStep(self):
+        volume=Volume()
+        volume.setFileName(self._getFileName(OUTPUT_RESOLUTION_FILE))
+
+        volume.setSamplingRate(self.inputVolume.get().getSamplingRate())
+        self._defineOutputs(resolution_Volume=volume)
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: Is this exactly the same as in the input? If so, we can skip this output.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+
+
+    # --------------------------- INFO functions ------------------------------
+
+    def _methods(self):
+        messages = []
+        if hasattr(self, 'resolution_Volume'):
+            messages.append(
+                'Information about the method/article in ' + MONORES_METHOD_URL)
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: `MONORES_METHOD_URL` is not defined here. You can import it as `from .protocol_resolution_monogenic_signal import MONORES_METHOD_URL`, and then use it.

File: xmipp3/protocols/protocol_resolution_localfilter.py [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+        return messages
+    
+    def _summary(self):
+        summary = []
+        summary.append("Highest resolution %.2f Å,   "
+                       "Lowest resolution %.2f Å. \n" % (self.min_res_init,
+                                                         self.max_res_init))
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_resolution_localfilter.py]: `self.min_res_init` and `self.max_res_init` are just initialized in line 55-56 but they are not updated anywhere.

**Comment by dmaluenda** [comment]
I undersand that the close of this PR is an error from Josue. I will reopen it!

**Comment by Vilax** [comment]
I did the pull request because we are in the middle of a review process.

Let me isolate the code.  I only need the resolution file, the rest of files will be finish soon and then I will do another pull request.

**Comment by dmaluenda** [comment]
That's fine for me, but it should work in some way. Fix, at least, the issues that crash the code and the rest will be fixed in the near future.

**Pull request closed without merging**